File: core/repository/intelligence/architecture_reconstruction_engine.py
from __future__ import annotations

def reconstruct_architecture(topology: dict, imports: list, routes: list, dependencies: list):
    services = sorted(set(topology.get('services', [])))
    db = sorted(set([d for d in (dependencies or []) if any(k in d.lower() for k in ['postgres','mysql','sqlite','mongo'])]))
    frontend = any(any(k in d.lower() for k in ['react','vue','angular','next']) for d in (dependencies or []))
    backend = any(any(k in d.lower() for k in ['flask','django','fastapi','express','spring']) for d in (dependencies or []))
    return {
      "services": services,
      "apis": sorted(set(routes or [])),
      "workers": [], "schedulers": [], "queues": [], "gateways": [],
      "databases": db,
      "frontend_backend_split": {"frontend": frontend, "backend": backend},
      "relationships": [{"from": services[i], "to": services[i+1]} for i in range(max(0,len(services)-1))]
    }

File: src/repository/intelligence/architectureReconstructionEngine.ts
/**
 * Converted from Python: core/repository/intelligence/architecture_reconstruction_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../../runtime/pyCompat.js";

export function reconstructArchitecture(topology: any, imports: any, routes: any, dependencies: any): any {
  var services: any = py.sorted(py.toSet(py.get(topology, "services", [])));
  var db: any = py.sorted(py.toSet(py.iter(py.or2(dependencies, () => ([]))).filter((d: any) => py.any(py.iter(["postgres", "mysql", "sqlite", "mongo"]).map((k: any) => py.contains(String(d).toLowerCase(), k)))).map((d: any) => d)));
  var frontend: any = py.any(py.iter(py.or2(dependencies, () => ([]))).map((d: any) => py.any(py.iter(["react", "vue", "angular", "next"]).map((k: any) => py.contains(String(d).toLowerCase(), k)))));
  var backend: any = py.any(py.iter(py.or2(dependencies, () => ([]))).map((d: any) => py.any(py.iter(["flask", "django", "fastapi", "express", "spring"]).map((k: any) => py.contains(String(d).toLowerCase(), k)))));
  return {"services": services, "apis": py.sorted(py.toSet(py.or2(routes, () => ([])))), "workers": [], "schedulers": [], "queues": [], "gateways": [], "databases": db, "frontend_backend_split": {"frontend": frontend, "backend": backend}, "relationships": py.range(py.max([0, py.sub(py.len(services), 1)])).map((i: any) => ({"from": py.at(services, i), "to": py.at(services, py.add(i, 1))}))};
}
